yaml2workflow/parser.py

```diff
@@ -54,26 +56,55 @@ def parse(filename: str, generate_new_id: bool = False, stub: service_pb2_grpc.V
     return [resources_pb2.Workflow(id=workflow_id, nodes=nodes)]
 
 
-def parse_model(model: typing.Dict, stub: service_pb2_grpc.V2Stub = None,
-                metadata=None) -> resources_pb2.Model:
-    model_id, model_version_id = search_model(model, stub=stub, metadata=metadata)
+def _parse_model(yaml_model: typing.Dict, stub: service_pb2_grpc.V2Stub = None,
+                 metadata=None) -> resources_pb2.Model:
+    model_id, model_version_id = _search_model(yaml_model, stub=stub, metadata=metadata)
 
     return resources_pb2.Model(
         id=model_id,
         model_version=resources_pb2.ModelVersion(id=model_version_id))
 
 
-def search_model(model: typing.Dict, stub: service_pb2_grpc.V2Stub = None,
-                 metadata=None) -> typing.Tuple[str, str]:
-    if 'model_id' in model and 'model_version_id' in model:
-        return model['model_id'], model['model_version_id']
+def _search_model(yaml_model: typing.Dict, stub: service_pb2_grpc.V2Stub = None,
+                  metadata=None) -> typing.Tuple[str, str]:
+    assert stub, "Stub is required to load model by ID."
+    assert metadata, "Metadata is required to load model by ID."
+    model_version_id = yaml_model['model_version_id'] if 'model_version_id' in yaml_model else None
```

Comment:
Perhaps simplify to this?
```suggestion
    model_version_id = yaml_model.get('model_version_id')
```